src/Tools/ExternalAccess/RazorTest/RazorPredefinedProviderNameTests.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Razor.UnitTests
+{
+    public class RazorPredefinedProviderNameTests
+    {
+        [Theory]
+        [InlineData(typeof(PredefinedCodeFixProviderNames), typeof(RazorPredefinedCodeFixProviderNames))]
+        [InlineData(typeof(PredefinedCodeRefactoringProviderNames), typeof(RazorPredefinedCodeRefactoringProviderNames))]
+        internal void RoslynNamesExistAndValuesMatchInRazorNamesClass(Type roslynProviderNamesType, Type razorProviderNamesType)
+        {
+            var roslynProviderNames = GetPredefinedNamesFromType(roslynProviderNamesType);
+            var razorProviderNames = GetPredefinedNamesFromType(razorProviderNamesType);
+
+            var failureMessage = new StringBuilder();
+            failureMessage.AppendLine($"The following Names were inconsistent between {roslynProviderNamesType.Name} and {razorProviderNamesType.Name}:");
+            var passLength = failureMessage.Length;
+
+            // Validates that all names from Roslyn's PredefinedCode*ProviderNames class exist in the RazorPredefinedCode*ProviderNames class
+            // and that they have the same value. It does not fail if the Razor class contains names no longer in the Roslyn class as they may
```

Comment:
> Validates that all names from Roslyn's PredefinedCode\*ProviderNames class exist in the RazorPredefinedCode\*ProviderNames

This is not a desirable characteristic (see prior comments).

> and that they have the same value

This is not a desirable characteristic. The property definitions explicitly allow these to decouple if/when necessary.

Reply:
> This is not a desirable characteristic (see prior comments).

I feel like the PR when a name is added is a good time to ensure Razor will be able to consume it. Razor still gets to choose when they make use of the new Name.

> This is not a desirable characteristic. The property definitions explicitly allow these to decouple if/when necessary.

I think it is useful to help Razor reason about which Provider is which.